workflows: Add e2e test for kube-apiserver HA

aditighag opened the pull request:
This PR adds an e2e test to verify that the cilium agent fails over to an active instance of the kube-apiserver when kube-proxy replacement is enabled. The new workflow deploys a kind cluster with multiple control plane nodes and no kube-proxy. 
All the kube-apiserver pods are then sequentially deleted to ensure that cilium agent pods correctly fail over - https://github.com/cilium/cilium/pull/37601.

aanm reviewed (changes requested):
Instead of create one more workflow, why don't we adjust one of the existing workflows to run k8s with 2 control planes and configure Cilium appropriately?

aditighag committed:
workflows: Add e2e test for kube-apiserver HA

This PR adds an e2e test to verify that the cilium
agent fails over to an active instance of the kube-apiserver
when kube-proxy replacement is enabled. It deploys a
kind cluster with multiple control plane nodes and kube-proxy
replacement is enabled with no kube-proxy.
All the kube-apiserver pods are then sequentially deleted to
ensure that cilium agent pods correctly fail over.

Signed-off-by: Aditi Ghag <aditi@cilium.io>

Artyop reviewed (approved):
lgtm, thank you :grin:
